src/v/kafka/server/handlers/produce.cc

```diff
@@ -439,9 +464,10 @@ produce_handler::handle(request_context&& ctx, ss::smp_service_group ssg) {
           error_code::transactional_id_authorization_failed));
 
     } else if (request.has_idempotent) {
-        return ctx.respond(request.make_error_response(
-          error_code::cluster_authorization_failed));
-
+        if (!config::shard_local_cfg().enable_idempotence.value()) {
```

Comment:
you might consider setting some feature flags on the `request` object when it is created instead of querying the configuration when its needed.

Reply:
good point. specially since this is in the hot path.